test/framework/src/main/java/org/elasticsearch/test/rest/client/RestTestClient.java

```diff
@@ -296,7 +303,17 @@ private static RestClient createRestClient(URL[] urls, Settings settings) throws
                 .register("http", PlainConnectionSocketFactory.getSocketFactory())
                 .register("https", sslsf)
                 .build();
-        CloseableHttpClient httpClient = RestClient.Builder.createDefaultHttpClient(socketFactoryRegistry);
+        PoolingHttpClientConnectionManager connectionManager = new PoolingHttpClientConnectionManager(socketFactoryRegistry);
+        //default settings may be too constraining
+        connectionManager.setDefaultMaxPerRoute(10);
+        connectionManager.setMaxTotal(30);
+
+        //default timeouts are all infinite
+        RequestConfig requestConfig = RequestConfig.custom().setConnectTimeout(CONNECT_TIMEOUT_MILLIS)
+            .setSocketTimeout(SOCKET_TIMEOUT_MILLIS)
+            .setConnectionRequestTimeout(CONNECTION_REQUEST_TIMEOUT_MILLIS).build();
+        CloseableHttpClient httpClient = HttpClientBuilder.create()
+            .setConnectionManager(connectionManager).setDefaultRequestConfig(requestConfig).build();
```

Comment:
I've also increased the connect and connectionrequest timeouts. I'm not sure if users may only want to increase socket timeout...